### ISSUE TITLE
sensuctl dump should support --format json

### ISSUE THREAD

**Issue by treydock** [sensu/sensu-go#3424]
## Expected Behavior

I'd expect `sensuctl dump` to support `--format json` like all other commands.

## Current Behavior

Currently only wrapped-json and YAML are supported and neither can be parsed by Ruby without hacking the output.

## Context

Utilizing `sensuctl dump` more in sensu-puppet and relying only on JSON would be nice.

## Your Environment

* Sensu version used (sensuctl, sensu-backend, and/or sensu-agent): 5.15
* Installation method (packages, binaries, docker etc.): RPM
* Operating System and version (e.g. Ubuntu 14.04): CentOS 7

**Comment by portertech** [comment]
@treydock what's the use case? `sensuctl dump` is intended for the backup/restore use case, hence wrapped objects.

**Comment by treydock** [comment]
In the https://github.com/sensu/sensu-puppet module we help users by having native Puppet types for managing Etcd replicators and PostgreSQL event storage configs and to do this we have to be able to create / update / list / delete using sensuctl. Right now the only way to "list" those resources is via `sensuctl dump`. We have relied heavily on `--format json` because the output could be directly passed to Ruby JSON for loading. Now we have to implement ugly hacks because `--format yaml` can't be loaded using Ruby YAML library without first splitting the output on `---` and then iterating over each element.

In the future the Puppet module will support and maybe move to using the API where possible but for now we do all management through sensuctl.

The "hacks" we've implemented work, but it would be nice to not need them hence the desire for `--format json`. If etcd replicators and postgresql configs had subcommands for list / delete then we could move away from using `dump`. We do all create / update through `sensuctl create -f`.
